src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  static addToAlive(animal) {
+    this.alive.push(animal);
+  }
+
+  static removeFromAlive(animal) {
+    if (this.alive.includes(animal)) {
+      const indexOfAnimal = this.alive.indexOf(animal);
+
+      this.alive.splice(indexOfAnimal);
+    }
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.addToAlive(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore) {
+      animal.health = animal.hidden ? animal.health : animal.health - 50;
+    }
+
+    if (animal instanceof Herbivore && animal.health === 0) {
```

Comment:
This check is redundant here. If an animal is dead, it doesn't matter it's herbivore or carnivore)
```suggestion
    if (animal.health === 0) {
```